src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let finalString = '';
+
+    if ((this[0] !== null)
+        && (this[0] !== undefined)) {
+      finalString += this[0];
+    }
+
+    for (let i = 1; i < this.length; i++) {
+      finalString += String(separator);
```

Comment:
```suggestion
    for (let i = 0; i < this.length; i++) {
     if (i) {
        finalString += String(separator);
     }
```

Reply:
Ok, but I think this isn`t better solution: we will do extra ckeck avery loop.